package.json

```diff
@@ -2,7 +2,6 @@
   "name": "react-dates",
   "version": "21.8.0",
   "description": "A responsive and accessible date range picker component built with React",
-  "main": "index.js",
```

Comment:
this would break every single consumer, including ESM ones, using the specifier `react-dates`

Reply:
Why, when it defaults to index?

Reply:
ah, that's a fair point, i never rely on the default so i forgot that.

That said, explicit > implicit, so altho this part isn't a breaking change, let's please revert it.

Reply:
Yes, but then it's not going to work, because it will always load the index.js file regardless of extension preference. Then it makes more sense to add the "exports" field instead.

Or perhaps revisit the idea of adding the "module" field? It's not standardized, but still widely used and very unlikely to break anything.

Reply:
I'm confused. The default is `index.js`, whether you import or require it.

Reply:
The default is index.something, node supports index.js, index.json, index.node, perhaps also something else. Webpack can be configured to prefer mjs over others, so it would use index.mjs.

But now thinking about it, it doesn't really sound like a good idea any more.

Reply:
That's not accurate - in CJS, extension resolution applies, but not in ESM. This means that `index.js` (in a package without type: module) is always CJS, and `import('react-dates')` continues to work, with or without it, by importing the CJS file.

The only resolution a package should expect is node's resolution. Using "exports", we can provide an ESM file, and then tools like webpack can use those as they like.